src/dataflow/src/source/file.rs

```diff
@@ -340,45 +343,54 @@ pub fn read_file_task<Ctor, I, Out, Err>(
             //
             // https://github.com/notify-rs/notify/issues/240
             #[cfg(target_os = "macos")]
-            let (file_events_stream, handle) = {
-                let (timer_tx, timer_rx) = std::sync::mpsc::channel();
-                std::thread::spawn(move || {
-                    while let Ok(()) = timer_tx.send(()) {
-                        std::thread::sleep(std::time::Duration::from_millis(100));
-                    }
-                });
-                (timer_rx, ())
-            };
+            thread::spawn(move || {
+                while let Ok(()) = notice_tx.send(()) {
+                    thread::sleep(std::time::Duration::from_millis(100));
+                }
+            });
 
-            #[cfg(not(target_os = "macos"))]
-            let (file_events_stream, handle) = {
-                let (notice_tx, notice_rx) = std::sync::mpsc::channel();
-                let mut w = match notify::RecommendedWatcher::new_raw(notice_tx) {
+            #[cfg(target_os = "linux")]
+            {
+                let mut inotify = match Inotify::init() {
                     Ok(w) => w,
                     Err(err) => {
-                        error!(
-                            "file source: failed to create notify watcher: {:#} (path: {})",
-                            err,
-                            path.display()
-                        );
+                        error!("file source: failed to initialize: {:#}", err,);
```

Comment:
Can we say something about inotify here? Just printout out the `std::io::Error` doesn't seem like enough context.

Reply:
Whoops, yeah, I meant to and just flubbed it. Fixed now.